### ISSUE TITLE
Feature suggestion: adding the option to download a torrent in sequential order

### ISSUE THREAD

**Issue by fad1** [transmission/transmission#597]
This is useful for large files that can be consumed as they are downloaded. i.e. to watch large video files (movies) while they are being downloaded. 

That's the one feature I miss switching from Qbittorrent.

**Comment by imanushin** [comment]
Is this PR related to feature https://github.com/transmission/transmission/pull/540 ?

**Comment by fad1** [comment]
Duplicate. #540

**Comment by andreygursky** [comment]
@mikedld commented:

> Might be easier to re-open #597 and mark this as invalid.
> On a side note, implementing this is unlikely to happen as it goes against the BitTorrent philosophy (e.g. see [Sequential downloading is bad](https://wiki.vuze.com/w/Sequential_downloading_is_bad)).

@fad1, feel free to reopen this issue.

@mikedld, I'm personally interested in this feature because I'm facing sometimes a situation, where there is only one seed appearing on an irregular basis and you don't know, whether he wouldn't disappear forever. There are no other peers. In this case, it makes more sense to have at least some consecutive part of data from the very beginning instead of many random chunks.

If some torrent has been just released, an activated super-seed mode should hopefully protect the seeder from giving away the same peace more than once(?) Then it is a decision of peers either to download something sequentially in say 80 minutes without "latency" or in say 15 minutes but with the latency of same 15 minutes. The only issue remains if not all peers choose the same strategy. Maybe some scientific paper exists that studies this in detail...

**Comment by andreygursky** [comment]
For reference:
- [TRAC #2632 Add streaming capability to libtransmission (but not the Transmission GUI clients)](https://trac.transmissionbt.com/ticket/2632)
- Added 'streaming' capability to libtransmission: https://github.com/cfpp2p/transmission/commit/e91d645d9a89c4591ef478e0979d558e8c70e90b

**Comment by mihaiav** [comment]
What is required for a good streaming experience  is random access not sequential order. The client may want to seek or the player itself may access different parts (i.e. to get meta data from the video container)

**Comment by ckerr** [comment]
https://github.com/transmission/transmission/wiki/Can-I-sequence-the-files-I-download

**Comment by AndreyPavlenko** [comment]
+1 to add support for this feature.
I use sequential download for a long time and it works just fine. https://github.com/AndreyPavlenko/transmission/commit/1f24be7a3211896cd2ad073ad7d83acb61452055.

**Comment by ckerr** [comment]
If someone were to PR an up-to-date version of this to `transmission/transmission` I'd be happy to review

**Comment by forteller** [comment]
Hi. I was pointed to this issue from my feature request #2208 

I tried explaining over there why it doesn't seem to me that these feature requests are related, but I never got a reply, so I'll try here instead. If I'm correct, I'd appreciate it if my issue could be opened again – and if I'm wrong I'd appreciate if anyone would take the time to explain to me how I'm wrong. I'm no expert, so I might be wrong! :)

Here's my thoughts:

Either I'm misunderstanding something here or I'm not explaining myself well enough (English is not my first language). Please let me try again, and see if you can tell me if I'm the one misunderstanding you, or if it's the other way around :)

The issue you linked to seems to me to be all about downloading the bits in a single file in order from first to last, so you can start watching a video file before the whole file is downloaded.

That is unimportant to me, and has nothing to do with what I'm trying to request, as far as I can tell (though I'm no technical expert in these things!)

What's important to me is that I can download the full first file in a torrent with many files in it first, then the full second file, etc. (Or the full first X files, then the next, but that's less important). This so that I can listen to the first part of an audioboook where each chapter is an individual file, or watch the first of a series of videos, without having to download the whole torrent first. And I would like to not have to manually go in and select the next file in the series every time the current one is downloaded.

It's very strange to me if this is fringe, I would think a lot of people would prefer to do this, even if they don't do it now because it's too much hassle if the torrent client doesn't automate the process.

This is especially important for people with low download speeds or torrents with relatively few seeders.

I did not see anything about any of this in the linked issue?

Thank you again!

**Comment by GScottElshaw** [comment]
I know it's not quite what you are looking for, I think, but why not just uncheck all of the files in the inspector, and then check the sequence you want them as you go?

**Comment by barhom** [comment]
+1, keeping me from using transmission right now :/

**Comment by rnhmjoj** [comment]
> But it seems this is very outdated, what is the new code change that's needed to be done and rebuild transmission?

I've been maintaining this [patch](https://maxwell.ydns.eu/~rnhmjoj/transmission-seq.patch). It works fine up to transmission 3.0.

**Comment by forteller** [comment]
> I know it's not quite what you are looking for, I think, but why not just uncheck all of the files in the inspector, and then check the sequence you want them as you go?

Sure, that is possible. Actually, that's what I do now. But it's quite annoying "manual labor" which I would think (though I'm not a coder) would be quite easy to automate, which is the whole point of my feature request. It's often possible to do something manually even though people ask for new features to make it simpler to do :)

**Comment by ckerr** [comment]
> > But it seems this is very outdated, what is the new code change that's needed to be done and rebuild transmission?
> 
> I've been maintaining this [patch](https://maxwell.ydns.eu/~rnhmjoj/transmission-seq.patch). It works fine up to transmission 3.0.

If you'd like to update this for 4.0.0 and put up a PR, I'd be happy to review it

**Comment by Pentaphon** [comment]
> I've been maintaining this [patch](https://maxwell.ydns.eu/~rnhmjoj/transmission-seq.patch). It works fine up to transmission 3.0.

@rnhmjoj you may want to do a PR for this now that 4.0.0 is out

**Comment by tulkki** [comment]
> > But it seems this is very outdated, what is the new code change that's needed to be done and rebuild transmission?
> 
> I've been maintaining this [patch](https://maxwell.ydns.eu/~rnhmjoj/transmission-seq.patch). It works fine up to transmission 3.0.

@rnhmjoj How did you apply this patch ?
I was looking for sequential order options but I only found old links.

**Comment by ckerr** [comment]
The patch will be in 4.1.0 but we're in a bugfix cycle right now so the release date for that version is TBD. For now you'd probably have to pull it from the PR and build it yourself, e.g. with https://github.com/transmission/transmission/pull/4795.patch
